### ISSUE TITLE
Handle missing data points better

### ISSUE THREAD

**Issue by bshor** [leeper/slopegraph#15]
Great package! But I'm still having difficulty with missing values. I'm attaching some data on state polarization for the northeast. Everything's there except the 2014 value for MA. This messes up the plot. I'd like to have the 2008 point for MA go directly to 2014. But as it is, MA gets orphaned after 2008, and the point for 2017 (the end) is not connected to anything.

This is the code I use:

```
rownames(sg)=sg$st; sg$st=NULL
colnames(sg)=str_c("Year.",colnames(sg))
    
slopegraph(sg, col.line='gray',col.lab = 'black', decimals=1,
     xlabels=c('1996','2000','2004','2008','2014','2017'), 
     cex.lab = 1, cex.num=0.75)
```

![ne polarization slopegraph](https://cloud.githubusercontent.com/assets/468070/25297001/bfa62f14-26b0-11e7-9b8e-94feac5b2dd1.png)
[slopegraph problem 042117.zip](https://github.com/leeper/slopegraph/files/948370/slopegraph.problem.042117.zip)

**Comment by ibecav** [comment]
Hi,

So let's assume that MA was missing in 2008 as you mentioned and that RI was missing in 2004 (it's less subtle and easy to see).  You want this straight line interpolation to happen correct?

![bbb](https://user-images.githubusercontent.com/26557145/42336424-a6a42664-8051-11e8-9605-6dc83404b923.png)

**Comment by bshor** [comment]
Thanks for taking this up!

I think the correct display should be not to impute a point (eg to display the numeric value) but rather to just draw the line from the last non-missing data value to the next non-missing data value.

For MA it should be a line from 2008 to 2017 with no numeric value present for 2014 (where there is missing data).

**Comment by ibecav** [comment]
Yes that's what it is doing in the example I gave you above. It's just difficult to see that because MA and NJ are so close together at that point.  It's easier to see in the RI example where I made 2004 = NA
